akka-typed-tests/src/test/java/akka/typed/javadsl/BehaviorChainTest.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.typed.javadsl;
+
+import akka.typed.Behavior;
+import akka.typed.Terminated;
+import org.junit.Test;
+import org.scalatest.junit.JUnitSuite;
+
+import java.util.ArrayList;
+
+import static akka.typed.javadsl.Actor.same;
+import static akka.typed.javadsl.Actor.stopped;
+
+public class BehaviorChainTest extends JUnitSuite {
+    interface Message {
+    }
+
+    static final class One implements Message {
+        public String foo() {
+          return "Bar";
+        }
+    }
+    static final class MyList<T> extends ArrayList<T> implements Message {
+    };
+
+    @Test
+    public void shouldCompile() {
+      Behavior<Message> b = Actor.<Message>immutable()
```

Comment:
`Actor.<Message>immutable` is a syntax that I think is rather unfamiliar for most Java devs. Without that explicit type the type inference doesn't work.

Here is an alternative:
```
def immutable2[T](b: JFunction[BehaviorChain[T], BehaviorChain[T]]): Behavior[T] = ???
```
that is used like:
```
      Behavior<Message> b = Actor.immutable2(b2 ->
        b2.message(One.class, (ctx, o) -> {
          o.foo();
          return same();
        })
        .message(One.class, o -> o.foo().startsWith("a"), (ctx, o) -> same())
      );
```

another advantage of that is that the builder passed as parameter and there is no question about how to create it, and it could be a real builder that we call `.build` on afterwards